cmd/thor/solo/solo.go

```diff
@@ -196,3 +204,53 @@ func (s *Solo) packing(pendingTxs tx.Transactions, onDemand bool) error {
 
 	return nil
 }
+
+// initSolo adds transactions to the first block to set the chain state
+func (s *Solo) initSolo() error {
+	gasPriceTx, err := s.makeGasPriceTx()
+	if err != nil {
+		return err
+	}
+
+	if err := s.txPool.Add(gasPriceTx); err != nil {
+		return err
+	}
+
+	return s.packing(tx.Transactions{gasPriceTx}, false)
+}
+
+// makeGasPriceTx creates a transaction to set the gas price
+func (s *Solo) makeGasPriceTx() (*tx.Transaction, error) {
```

Comment:
Should be `makeBaseGasPriceTx`,  this function is only used once, is it worth doing in a new function?